src/test/java/org/opentripplanner/middleware/bugsnag/BugsnagTest.java

```diff
@@ -1,67 +1,299 @@
+                    aResponse()
+                        .withBodyFile("createEventDataRequest.json")
+                )
+        );
+        BugsnagEventRequestJob.triggerEventDataRequestForProject(projectId, 14);
+        BugsnagEventRequest bugsnagEventRequest =
+            Persistence.bugsnagEventRequests.getOneFiltered(Filters.eq("projectId", projectId));
+        assertNotNull(bugsnagEventRequest);
+        bugsnagEventRequest.delete();
+    }
+
+    /**
+     * Provide a 'pending' event request which when refreshed (using CompletedEventDataRequest.son) is updated to
+     * 'completed'. Using the url parameter from the refreshed event request, process the event data response
+     * (eventDataResponse.json) and save the Bugsnag event. Confirm that the refreshed event request status has been
+     * updated to 'completed' and that the Bugsnag event has been saved.
+     *
+     * Note: the CompletedEventDataRequest.son contains a hardcoded reference to the wiremock URL:
+     * http://localhost:8089/bugsnag-event-data-requests/611a9123b6e99d4fcec2dc5c
+     */
+    @Test
+    public void processCompletedEventRequest() {
+        String projectId = "5ee8f4026c0a34000e1b1394"; // Must match the project id value in eventDataResponse.json
+        String eventDataRequestId = "611a9123b6e99d4fcec2dc5c"; // Must match the id value in createEventDataRequest.json
+        BugsnagEventRequest bugsnagEventRequest =
+            createBugsnagEventRequest(
+                projectId,
+                eventDataRequestId,
+                "pending",
+                getDateInPast(1)
+            );
+        // create wiremock stub for the refresh event data request.
+        wireMockServer.stubFor(
+            get(urlMatching(String.format("/projects/%s/event_data_requests/%s", projectId, eventDataRequestId)))
+                .willReturn(
+                    aResponse()
+                        .withBodyFile("completedEventDataRequest.json")
+                )
+        );
+        // create wiremock stub for the event data response.
+        wireMockServer.stubFor(
+            get(urlMatching(String.format("/bugsnag-event-data-requests/%s", eventDataRequestId)))
+                .willReturn(
+                    aResponse()
+                        .withBodyFile("eventDataResponse.json")
+                )
+        );
+        BugsnagEventHandlingJob bugsnagEventHandlingJob = new BugsnagEventHandlingJob();
+        bugsnagEventHandlingJob.refreshEventRequest(bugsnagEventRequest);
+        BugsnagEventRequest updatedBugsnagEventRequest = Persistence.bugsnagEventRequests.getById(bugsnagEventRequest.id);
+        assertEquals(updatedBugsnagEventRequest.status, "COMPLETED");
+        bugsnagEventRequest.delete();
+        BugsnagEvent bugsnagEvent = Persistence.bugsnagEvents.getOneFiltered(Filters.eq("projectId", projectId));
+        assertNotNull(bugsnagEvent);
+        Persistence.bugsnagEvents.removeById(bugsnagEvent.id);
```

Comment:
This seems like some kind of post-test cleanup. It might make sense to put this in a `@BeforeEach`-decorated method and always remove all bugsnagEvents/bugsnagEventRequests. This way if a single test fails before hitting the cleanup section, it can be ensured that the database will be good to go. This way tests can be certain to not have effects on other tests.